compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -223,6 +223,50 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         }
     }
 
+    fn switch_with_weights(
+        &mut self,
+        v: Self::Value,
+        else_llbb: Self::BasicBlock,
+        else_is_cold: bool,
+        cases: impl ExactSizeIterator<Item = (u128, Self::BasicBlock, bool)>,
+    ) {
+        if self.cx.sess().opts.optimize == rustc_session::config::OptLevel::No {
+            self.switch(v, else_llbb, cases.map(|(val, dest, _)| (val, dest)));
+            return;
+        }
+
+        let id_str = "branch_weights";
+        let id = unsafe {
+            llvm::LLVMMDStringInContext2(
+                self.cx.llcx,
+                id_str.as_ptr() as *const c_char,
+                id_str.len() as size_t,
```

Comment:
```suggestion
                id_str.len(),
```
This cast should always be unnecessary, since in Rust `usize` is essentially required to be `size_t`. Certainly this is the case on platforms we support as hosts... right? If you know different please do correct me.

Reply:
Changed